types/ember__application/index.d.ts

```diff
@@ -4,10 +4,133 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-import Ember from 'ember';
+import Engine from '@ember/engine';
+import ApplicationInstance from '@ember/application/instance';
+import EventDispatcher from '@ember/application/-private/event-dispatcher';
+import { EventDispatcherEvents } from '@ember/application/types';
+import DefaultResolver from '@ember/application/-private/default-resolver';
+import { Router } from '@ember/routing';
+import Registry from '@ember/application/-private/registry';
 
-export default class Application extends Ember.Application { }
-export const getOwner: typeof Ember.getOwner;
-export const onLoad: typeof Ember.onLoad;
-export const runLoadHooks: typeof Ember.runLoadHooks;
-export const setOwner: typeof Ember.setOwner;
+/**
+ * An instance of Ember.Application is the starting point for every Ember application. It helps to
+ * instantiate, initialize and coordinate the many objects that make up your app.
+ */
+export default class Application extends Engine {
+    /**
+     * Call advanceReadiness after any asynchronous setup logic has completed.
+     * Each call to deferReadiness must be matched by a call to advanceReadiness
+     * or the application will never become ready and routing will not begin.
+     */
+    advanceReadiness(): void;
+    /**
+     * Use this to defer readiness until some condition is true.
+     *
+     * This allows you to perform asynchronous setup logic and defer
+     * booting your application until the setup has finished.
+     *
+     * However, if the setup requires a loading UI, it might be better
+     * to use the router for this purpose.
+     */
+    deferReadiness(): void;
+    /**
+     * defines an injection or typeInjection
+     */
+    inject(factoryNameOrType: string, property: string, injectionName: string): void;
+    /**
+     * This injects the test helpers into the window's scope. If a function of the
+     * same name has already been defined it will be cached (so that it can be reset
+     * if the helper is removed with `unregisterHelper` or `removeTestHelpers`).
+     * Any callbacks registered with `onInjectHelpers` will be called once the
+     * helpers have been injected.
+     */
+    injectTestHelpers(): void;
+    /**
+     * registers a factory for later injection
+     * @param fullName type:name (e.g., 'model:user')
+     * @param factory (e.g., App.Person)
+     */
+    register(fullName: string, factory: any): void;
```

Comment:
https://emberjs.com/api/ember/3.4/classes/Application/methods/register?anchor=register

The 3rd parameter is missing here?

Reply:
This PR is just moving types around. The issue you're pointing out is correct, but also present in the current types

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/adf4e18eef97454fab36830221b9b21c517af004/types/ember/index.d.ts#L365-L370

Please report this as a seperate issue here https://github.com/typed-ember/ember-cli-typescript/issues/new?template=TYPES_BUG.md&labels=%5Btypes,bug%5D&title=%5B@types/ember%20bug%5D%20-%20%3CYOUR_DESCRIPTION_HERE%3E